Lib/functools.py

```diff
@@ -284,7 +284,7 @@ def __new__(cls, func, /, *args, **keywords):
         if not callable(func):
             raise TypeError("the first argument must be callable")
 
-        if hasattr(func, "func"):
+        if isinstance(func, partial) and hasattr(func, "func"):
```

Comment:
Will an instance of `partial` ever not have a `func` attribute? Could this be simplified to just an `isinstance()` check?

```suggestion
        if isinstance(func, partial):
```

Reply:
yes, that works for me too. (you can certainly remove the `func` attribute from a pure python `partial` instance, but the resulting object is not very useful)

Reply:
if we do that, maybe it's also a good idea to just use an isinstance check in the C implementation as well? then the two approaches are straightforwardly identical.

Reply:
> you can certainly remove the func attribute from a pure python partial instance, but the resulting object is not very useful

Yeah, if people are deleting the `func` attribute from `partial` objects, I have a lot of questions about their broader approach to programming in general :)

> maybe it's also a good idea to just use an isinstance check in the C implementation as well?

I'm not qualified to comment on the C implementation, so I'll defer to you and @rhettinger on that question :)

Reply:
> Yeah, if people are deleting the func attribute from partial objects, I have a lot of questions about their broader approach to programming in general :)

this argument convinced me ;-). I just went with the `isinstance` check in both places.